File: app/src/main/java/fr/paug/androidmakers/util/TimeUtils.java
package fr.paug.androidmakers.util;

import android.content.Context;

import java.text.DateFormat;
import java.util.Date;
import java.util.TimeZone;

import androidx.annotation.NonNull;
import fr.paug.androidmakers.R;

public class TimeUtils {

    public static String dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ";

    public static final int SECOND = 1000;
    public static final int MINUTE = 60 * SECOND;
    public static final int HOUR = 60 * MINUTE;
    public static final int DAY = 24 * HOUR;

    public static String formatShortTime(Context context, Date time) {
        // Android DateFormatter will honor the user's current settings.
        DateFormat dateFormat = android.text.format.DateFormat.getTimeFormat(context);
        // Override with Timezone based on settings since users can override their phone's timezone
        // with Pacific time zones.
        TimeZone tz = TimeZone.getDefault();
        if (tz != null) {
            dateFormat.setTimeZone(tz);
        }
        return dateFormat.format(time).toLowerCase();
    }

    public static String formatDuration(@NonNull Context context, long startTime, long endTime) {
        return formatDuration(context, endTime - startTime);
    }

    private static String formatDuration(@NonNull Context context, long duration) {
        Float hours = duration / (float) HOUR;
        if (hours >= 1f) {
            return context.getResources().getQuantityString(R.plurals.duration_hours,
                    (int) Math.ceil(hours),
                    (hours == hours.intValue()) ? String.valueOf(hours.intValue()) : String.valueOf(Math.round(hours * 100d)/100d));
        } else {
            long minutes = duration / MINUTE;
            return context.getResources().getQuantityString(
                    R.plurals.duration_minutes,
                    (int) minutes,
                    minutes);
        }
    }

}

File: app/src/main/java/fr/paug/androidmakers/util/TimeUtils.kt
package fr.paug.androidmakers.util

import android.content.Context
import android.text.format.DateFormat
import fr.paug.androidmakers.R
import java.time.OffsetDateTime
import java.time.ZoneId
import java.util.Date
import java.util.Locale
import java.util.TimeZone
import kotlin.time.Duration

object TimeUtils {
  const val SECOND = 1000
  const val MINUTE = 60 * SECOND
  const val HOUR = 60 * MINUTE

  fun parseIso8601(iso8601: String): Date {
    return Date(OffsetDateTime.parse(iso8601).toEpochSecond() * 1000)
  }

  /**
   * format the time as HH:mm in the timezone of the event
   */
  fun formatShortTime(context: Context?, time: Date?): String {

    // Android DateFormatter will honor the user's current settings.
    val dateFormat = DateFormat.getTimeFormat(context)

    // Display the time in the local time as everyone will be on site so it makes
    // planning in advance easier
    val tz = TimeZone.getTimeZone("Europe/Paris")
    if (tz != null) {
      dateFormat.timeZone = tz
    }
    return dateFormat.format(time).lowercase(Locale.getDefault())
  }

  fun formatDuration(context: Context, duration: Duration): String {
    val minutes = duration.inWholeMinutes
    return if (minutes <= 120) {
      context.resources.getQuantityString(
          R.plurals.duration_minutes, minutes.toInt(),
          minutes
      )
    } else {
      context.resources.getString(R.string.many_minutes)
    }
  }
}

val eventTimeZone = kotlinx.datetime.TimeZone.of("Europe/Paris")